src/plugins/unified_search/public/search_bar/search_bar.tsx

```diff
@@ -42,572 +42,426 @@ export interface SearchBarInjectedDeps {
+    if (!showDatePicker && dateRangeFrom !== undefined && dateRangeTo !== undefined) {
+      return false;
     }
-
-    let nextDateRange = null;
-    if (
-      nextProps.dateRangeFrom !== get(prevState, 'currentProps.dateRangeFrom') ||
-      nextProps.dateRangeTo !== get(prevState, 'currentProps.dateRangeTo')
-    ) {
-      nextDateRange = {
-        dateRangeFrom: nextProps.dateRangeFrom,
-        dateRangeTo: nextProps.dateRangeTo,
-      };
+    if (indexPatterns?.length) {
+      // return true if at least one of the DateView has timeFieldName
+      return indexPatterns.some((dataView) => Boolean(dataView.timeFieldName));
     }
+    return true;
+  };
 
-    const nextState: any = {
-      currentProps: nextProps,
-    };
-    if (nextQuery) {
-      nextState.query = nextQuery;
-    }
-    if (nextDateRange) {
-      nextState.dateRangeFrom = nextDateRange.dateRangeFrom;
-      nextState.dateRangeTo = nextDateRange.dateRangeTo;
-    }
-    return nextState;
-  }
-
-  /*
-   Keep the "draft" value in local state until the user actually submits the query. There are a couple advantages:
-
-    1. Each app doesn't have to maintain its own "draft" value if it wants to put off updating the query in app state
-    until the user manually submits their changes. Most apps have watches on the query value in app state so we don't
-    want to trigger those on every keypress. Also, some apps (e.g. dashboard) already juggle multiple query values,
-    each with slightly different semantics and I'd rather not add yet another variable to the mix.
-
-    2. Changes to the local component state won't trigger an Angular digest cycle. Triggering digest cycles on every
-    keypress has been a major source of performance issues for us in previous implementations of the query bar.
-    See https://github.com/elastic/kibana/issues/14086
-  */
-  public state = {
-    isFiltersVisible: true,
-    openQueryBarMenu: false,
-    showSavedQueryPopover: false,
-    currentProps: this.props,
-    query: this.props.query ? { ...this.props.query } : undefined,
-    dateRangeFrom: get(this.props, 'dateRangeFrom', 'now-15m'),
-    dateRangeTo: get(this.props, 'dateRangeTo', 'now'),
-  } as State<QT>;
-
-  public isDirty = () => {
-    if (!this.props.showDatePicker && this.state.query && this.props.query) {
-      return !isEqual(this.state.query, this.props.query);
+  const isDirty = () => {
+    if (!showDatePicker && stateQuery && query) {
+      return !isEqual(stateQuery, query);
     }
 
     return (
-      (this.state.query && this.props.query && !isEqual(this.state.query, this.props.query)) ||
-      this.state.dateRangeFrom !== this.props.dateRangeFrom ||
-      this.state.dateRangeTo !== this.props.dateRangeTo
+      (stateQuery && query && !isEqual(stateQuery, query)) ||
+      stateDateRangeFrom !== dateRangeFrom ||
+      stateDateRangeTo !== dateRangeTo
     );
   };
 
-  componentWillUnmount() {
-    this.renderSavedQueryManagement.clear();
```

Comment:
💬 This `.clear` was being called on an instance of a `memoizeOne`'d component. Now using `React.memo` instead so don't think this is still needed.